### PR TITLE
PC-1455

### PR DESCRIPTION
**Pull request by eyalvo-matrix**
review it

### CODE REVIEW

File: modules/FhirAPI/src/FhirAPI/Service/FhirRequestParamsHandler.php [hunk 1]
```diff
@@ -128,6 +129,17 @@ private function processRequestParams(){
                                 $requestParams['PARAMETERS_FOR_SEARCH_RESULT'][$name][] =$tempValue;
                             }
                             break;
+                        case "filter":
```
Comment (eyalvo-matrix) [inline modules/FhirAPI/src/FhirAPI/Service/FhirRequestParamsHandler.php:132]: http://localhost:8565/apis//fhir/v4/ValueSet/gender/$expand?filter=^FeM => like search
http://localhost:8565/apis//fhir/v4/ValueSet/gender/$expand?filter=male   => exact

File: modules/FhirAPI/src/FhirAPI/Service/FhirRequestParamsHandler.php [hunk 2]
```diff
@@ -128,6 +129,17 @@ private function processRequestParams(){
...
+                            $cleanValue = preg_replace("/[\W]/", "", $value);
+                            $operators=  preg_replace("/[\w]/", "", $value);
+                            if(strpos($operators, '^') !== false){
```
Comment (matrix-amiel) [inline modules/FhirAPI/src/FhirAPI/Service/FhirRequestParamsHandler.php:135]: where do you found the ^ operator is regular like? (I've not fount it) usually ^ equal to '%string' and not  '%string%', prefer to check again...

File: modules/FhirAPI/src/FhirAPI/Service/FhirRequestParamsHandler.php [hunk 3]
```diff
@@ -34,7 +34,8 @@ class FhirRequestParamsHandler
                                               '_total',
                                               '_elements',
                                               '_contained',
-                                              '_containedType'
+                                              '_containedType',
+                                              'filter'
```
Comment (matrix-amiel) [inline modules/FhirAPI/src/FhirAPI/Service/FhirRequestParamsHandler.php:38]: What the meaning that "filter" is in this list? is it removed from the regular query parameters? 
I'm not sure there aren't other resources with "filter" in the query string

I think it needs to be special parameter only for READ calls and nor for search

File: modules/FhirAPI/src/FhirAPI/Service/FhirRequestParamsHandler.php (hunk 2 above)
Comment (eyalvo-matrix) [reply modules/FhirAPI/src/FhirAPI/Service/FhirRequestParamsHandler.php:135]: I can not use % I am open to any other char ....

File: modules/FhirAPI/src/FhirAPI/Service/FhirRequestParamsHandler.php (hunk 3 above)
Comment (eyalvo-matrix) [reply modules/FhirAPI/src/FhirAPI/Service/FhirRequestParamsHandler.php:38]: it is not removed just handled if you know where we use filter I can check it is not hurt

Comment (eyalvo-matrix) [reply modules/FhirAPI/src/FhirAPI/Service/FhirRequestParamsHandler.php:38]: image

Comment (eyalvo-matrix) [reply modules/FhirAPI/src/FhirAPI/Service/FhirRequestParamsHandler.php:38]: for others we can use "_filter"   about just filter I don't know it is not in any list.

**Comment by eyalvo-matrix** [comment]
hi

**Pull request merged**